agent/app/agent_unix.go

```diff
@@ -162,6 +163,7 @@ func (agent *ecsAgent) initializeResourceFields(credentialsManager credentials.M
 		},
 		Ctx:              agent.ctx,
 		DockerClient:     agent.dockerClient,
+		S3ClientCreator:  s3factory.NewS3ClientCreator(),
```

Comment:
since both Windows and Linux implementations use this, we can move it to `ResourceFieldsCommon`: https://github.com/aws/amazon-ecs-agent/blob/master/agent/taskresource/types_common.go#L24

Reply:
moved to common resource fields